core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java

```diff
@@ -65,19 +70,51 @@ class LineTcpConnectionContext extends AbstractMutableIOContext<LineTcpConnectio
         recvBufStart = Unsafe.malloc(configuration.getNetMsgBufferSize(), MemoryTag.NATIVE_ILP_RSS);
         recvBufEnd = recvBufStart + configuration.getNetMsgBufferSize();
         clear();
+        this.maintenanceInterval = configuration.getMaintenanceInterval();
+        this.maintenanceJobDeadline = milliClock.getTicks() + maintenanceInterval;
+        this.nextCommitTime = milliClock.getTicks();
     }
 
     @Override
     public void clear() {
         recvBufPos = recvBufStart;
         peerDisconnected = false;
         resetParser();
+        ObjList<ByteCharSequence> keys = tableUpdateDetailsUtf8.keys();
+        for (int n = keys.size() - 1; n >= 0; --n) {
+            final ByteCharSequence tableNameUtf8 = keys.get(n);
+            final TableUpdateDetails tud = tableUpdateDetailsUtf8.get(tableNameUtf8);
+            tud.releaseWriter(true);
```

Comment:
it looks as if writer can be released from `tud.close()` there is also a place where `tud` is closed but writer not released:

Reply:
fixed